src/routes/safe/components/Settings/SpendingLimit/NewLimitModal/Review.tsx

```diff
@@ -0,0 +1,221 @@
+import { Button, Text } from '@gnosis.pm/safe-react-components'
+import React from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+
+import Block from 'src/components/layout/Block'
+import Col from 'src/components/layout/Col'
+import Row from 'src/components/layout/Row'
+import { getNetworkInfo } from 'src/config'
+import {
+  getGnosisSafeInstanceAt,
+  getSpendingLimitContract,
+  MULTI_SEND_ADDRESS,
+} from 'src/logic/contracts/safeContracts'
+import createTransaction from 'src/logic/safe/store/actions/createTransaction'
+import { SpendingLimit } from 'src/logic/safe/store/models/safe'
+import { safeParamAddressFromStateSelector, safeSpendingLimitsSelector } from 'src/logic/safe/store/selectors'
+import { DELEGATE_CALL, TX_NOTIFICATION_TYPES } from 'src/logic/safe/transactions'
+import { getEncodedMultiSendCallData, MultiSendTx } from 'src/logic/safe/utils/upgradeSafe'
+import { Token, makeToken } from 'src/logic/tokens/store/model/token'
+import { ZERO_ADDRESS } from 'src/logic/wallets/ethAddresses'
+import { getWeb3 } from 'src/logic/wallets/getWeb3'
+import { ActionCallback, CREATE } from 'src/routes/safe/components/Settings/SpendingLimit/NewLimitModal/index'
+import { SPENDING_LIMIT_MODULE_ADDRESS } from 'src/utils/constants'
+
+import { RESET_TIME_OPTIONS } from 'src/routes/safe/components/Settings/SpendingLimit/FormFields/ResetTime'
+import { AddressInfo, ResetTimeInfo, TokenInfo } from 'src/routes/safe/components/Settings/SpendingLimit/InfoDisplay'
+import Modal from 'src/routes/safe/components/Settings/SpendingLimit/Modal'
+import { useStyles } from 'src/routes/safe/components/Settings/SpendingLimit/style'
+import {
+  adjustAmountToToken,
+  currentMinutes,
+  SpendingLimitRow,
+} from 'src/routes/safe/components/Settings/SpendingLimit/utils'
+import { toTokenUnit, fromTokenUnit } from 'src/logic/tokens/utils/humanReadableValue'
+
+const { nativeCoin } = getNetworkInfo()
+
+interface ReviewSpendingLimitProps {
+  onBack: ActionCallback
+  onClose: () => void
+  txToken: Token
+  values: Record<string, string>
+  existentSpendingLimit?: SpendingLimitRow
+}
+
+const Review = ({ onBack, onClose, txToken, values }: ReviewSpendingLimitProps): React.ReactElement => {
+  const classes = useStyles()
+
+  const dispatch = useDispatch()
+
+  const safeAddress = useSelector(safeParamAddressFromStateSelector)
+  const spendingLimits = useSelector(safeSpendingLimitsSelector)
+
+  // undefined: before setting a value
+  // null: if no previous value
+  // SpendingLimit: if previous value exists
+  const [existentSpendingLimit, setExistentSpendingLimit] = React.useState<SpendingLimit | null>(null)
+  React.useEffect(() => {
+    const checkExistence = async () => {
```

Comment:
Would be possible to move all this logic to a hook?

Reply:
I'm not sure if it's worth the move.

If it's for code organization, I would prefer to rethink the `handleSubmit` callback instead.

I'll leave this open to discussion tho.

Reply:
Yes I think it would be easier to follow, improving `handleSubmit` logic would be a good idea too, anyway, if you think it's too complicate and it's not worth just leave it